code/script.js

```diff
@@ -301,25 +415,86 @@ const filterCharacters = (keep) => {
   */
 
   // Invoke a function to redraw the board with the remaining people.
-}
+ 
+
+ const filteredCharacters = CHARACTERS.filter( (singleCharacter) => {
+ if (singleCharacter.hair === 'black') {
+   return singleCharacter;
+ }  
+});
+
+filteredCharacters.forEach((charachter)=>  {
+charachter.hair = 'yellow'  
+})
+
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
 
+
+const guess = (personToConfirm) => {      // store the interaction from the player in a variable.
+
+  let confirmation = confirm (`Are you sure that ${personToConfirm} is the secret person? `) // remember the confirm() ?
+  if (confirmation === true)
+{
+  checkMyGuess (personToConfirm)
+}   
+}
+// If the player wants to guess, invoke the checkMyGuess function.
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+
+  if (personToCheck === secret.name) {    // 1. Check if the personToCheck is the same as the secret person's name
+    winOrLoseText.innerHTML (`You are a WINNER!`)
+ } else {
+    winOrLoseText.innerHTML (`I'm sorry! The secret person is ${secret.name}`)// 2. Set a Message to show in the win or lose section accordingly
```

Comment:
In order to make this work there should be an equal sign after the innerHTML otherwise it will be seen as a function which its not. Its the same for row 447.

`winOrLoseText.innerHTML = (`I'm sorry! The secret person is ${secret.name}`)`